fetch.bs

```diff
@@ -6655,6 +6697,239 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+ "{{NotAllowedError}}" {{DOMException}}.
+
+ <li><p>Let <var>totalRequestLength</var> be the [=string/length=] of <var>request</var>'s
+ <a for=request>URL</a>, <a lt="URL serializer">serialized</a> with
+ [=URL serializer/exclude fragment=] set to true.
+
+ <li><p><a for=list>For each</a> (<var>name</var>, <var>value</var>) in
+ <a for=request>header list</a>, increment <var>totalRequestLength</var> by <var>name</var>'s
+ [=byte sequence/length=] + <var>value</var>'s [=byte sequence/length=].
+
+ <li>
+  <p>If <var>request</var>'s <a for=request>body</a> is non-null then:
+
+  <ol>
+   <li><p>If <var>request</var>'s
+   <a for=request>body</a>'s <a for=body>length</a> is null, then throw a {{TypeError}}.
+
+   <li>
+    <p>If <var>request</var>'s <a for=request>body</a>'s <a for=body>source</a> is null, then throw a
+    {{TypeError}}.
+
+    <p class=note>This disallows sending deferred fetches with a live {{ReadableStream}}.
+
+   <li><p>Increment <var>totalRequestLength</var> by <var>request</var>'s
+   <a for=request>body</a>'s <a for=body>length</a>.
+  </ol>
+
+ <li><p>If the <a>available deferred fetching quota</a> given <var>request</var>'s
+ <a for=request>client</a> and <var>request</var>'s <a for=request>URL</a>'s <a for=url>origin</a>
+ is less than <var>totalRequestLength</var>, then throw a "{{QuotaExceededError}}" {{DOMException}}.
+
+ <li><p>Set <var>request</var>'s <a for=request>service-workers mode</a> to "<code>none</code>".
+
+ <li><p>Set <var>request</var>'s <a for=request>keepalive</a> to true.
+
+ <li><p>Let <var>deferredRecord</var> be a new <a>deferred fetch record</a> whose
+ <a for="deferred fetch record">request</a> is <var>request</var>.
+
+ <li><p><a for=list>Append</a> <var>deferredRecord</var> to <var>request</var>'s
+ <a for=request>client</a>'s <a for=fetch>fetch group</a>'s
+ <a for="fetch group">deferred fetch records</a>.
+
+ <li>
+  <p>If <var>activateAfter</var> is non-null, then run the following steps <a>in parallel</a>:</p>
+
+  <ol>
+   <li><p>The user agent should wait until <var>activateAfter</var> milliseconds have passed,
+   or until the user agent has a reason to believe that it is about to lose the opportunity to
+   execute scripts, e.g., when the browser is moved to the background, or when <var>request</var>'s
+   <a for=request>client</a> is a <code>Document</code> that had a "<code>hidden</code>"
+   <a for=Document>visibility state</a> for a long period of time.
+
+   <li><p>The user agent may wait for a further <a>implementation-defined</a> <a>duration</a>, e.g.,
+   in order to batch several requests together or to wait until <a for=request>keepalive</a>
+   requests are complete.
+
+   <li><p><a>Process a deferred fetch</a> given <var>deferredRecord</var>.
+  </ol>
+ </li>
+
+ <li><p>Return <var>deferredRecord</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To get the <dfn>available deferred fetching quota</dfn> given a <code>Document</code>
```

Comment:
> To get the available deferred fetching quota 

It looks like this algorithm assumes that both `remainingTotalQuota` and `remainingQuotaForOrigin` are always >= 0, which may not be true within Step 6.